overlord/fdestate/fdemgr_test.go

```diff
@@ -134,7 +142,30 @@ func (s *fdeMgrSuite) startedManager(c *C) *fdestate.FDEManager {
 
 func (s *fdeMgrSuite) TestGetManagerFromState(c *C) {
 	st := s.st
-	manager := s.startedManager(c)
+	const onClassic = true
+	s.AddCleanup(release.MockOnClassic(onClassic))
+	manager := s.startedManager(c, onClassic)
+
+	st.Lock()
+	defer st.Unlock()
+	foundManager := fdestate.FdeMgr(st)
+	c.Check(foundManager, Equals, manager)
+
+	var fdeSt fdestate.FdeState
+	err := st.Get("fde", &fdeSt)
+	c.Assert(err, IsNil)
+	primaryKey, hasPrimaryKey := fdeSt.PrimaryKeys[0]
+	c.Assert(hasPrimaryKey, Equals, true)
+	c.Check(crypto.Hash(primaryKey.Digest.Algorithm), Equals, crypto.Hash(crypto.SHA256))
+	c.Check(primaryKey.Digest.Salt, DeepEquals, []byte{1, 2, 3, 4})
+	c.Check(primaryKey.Digest.Digest, DeepEquals, []byte{5, 6, 7, 8})
+}
+
+func (s *fdeMgrSuite) TestGetManagerFromStateWithWritable(c *C) {
```

Comment:
Nitpick, both tests seem identical, maybe the could be merged

```suggestion
func (s *fdeMgrSuite) testGetManagerFromState(c *C, onClassic bool)
	st := s.st
	defer release.MockOnClassic(onClassic)()
	manager := s.startedManager(c, onClassic)

	st.Lock()
	defer st.Unlock()
	foundManager := fdestate.FdeMgr(st)
	c.Check(foundManager, Equals, manager)
	var fdeSt fdestate.FdeState
	err := st.Get("fde", &fdeSt)
	c.Assert(err, IsNil)
	primaryKey, hasPrimaryKey := fdeSt.PrimaryKeys[0]
	c.Assert(hasPrimaryKey, Equals, true)
	c.Check(crypto.Hash(primaryKey.Digest.Algorithm), Equals, crypto.Hash(crypto.SHA256))
	c.Check(primaryKey.Digest.Salt, DeepEquals, []byte{1, 2, 3, 4})
	c.Check(primaryKey.Digest.Digest, DeepEquals, []byte{5, 6, 7, 8})
}

func (s *fdeMgrSuite) TestGetManagerFromState(c *C) {
    	const onClassic = true
    	s.testGetManagerFromState(c, onClassic)
}

func (s *fdeMgrSuite) TestGetManagerFromStateWithWritable(c *C) {
    	const onClassic = false
    	s.testGetManagerFromState(c, onClassic)
}

```